### PR TITLE
Updates to the Ethereum bridge spec

### PR DESCRIPTION
**Pull request by james-chf**
Small changes to the Ethereum bridge spec based on recent discussions

### CODE REVIEW

File: documentation/spec/src/interoperability/ethereum-bridge.md [hunk 1]
```diff
@@ -49,21 +48,26 @@ pub enum EthereumEvent {
...
-considered a slashable offence.
+Each event will be stored with a list of the validators that have ever seen it 
+as well as the fraction of total voting power that has ever seen it. 
+Once an event has been seen by 2/3 of voting power, it is locked into a
+`seen` state, and acted upon.
+
+There is no adjustment across epoch boundaries - e.g. if an event is seen by 1/3
+of voting power in epoch n, then seen by a different 1/3 of voting power in 
+epoch m>n, the event will be considered `seen` in total. Validators may never
+vote more than once for a given event.
```
Comment (james-chf) [inline documentation/spec/src/interoperability/ethereum-bridge.md:59]: TBD - https://github.com/anoma/namada/issues/181

File: documentation/spec/src/interoperability/ethereum-bridge.md [hunk 2]
```diff
@@ -49,21 +48,26 @@ pub enum EthereumEvent {
...
+### Minimum confirmations
+There will be a protocol-specified minimum number of confirmations that events
+must reach on the Ethereum chain, before validators should vote to include them
+on Namada. Validators should not vote to include events that have not met this 
+required number of confirmations. This constant will be changeable via 
+governance. Voting on unconfirmed events is considered a slashable offence.
+
+In a future version of the Ethereum bridge, we may allow an additional per-event
+minimum number of confirmations that an event should have to reach before being
+included in Namada.
```
Comment (james-chf) [inline documentation/spec/src/interoperability/ethereum-bridge.md:70]: Currently our Ethereum events don't include a custom additional minimum number of confirmations, I think we should do this as an extension once we have protocol-level minimum number of confirmations working

File: documentation/spec/src/interoperability/ethereum-bridge.md [hunk 3]
```diff
@@ -93,16 +98,19 @@ removed by the ledger code for the specific permitted transactions that are
...
 1. The storage value `/eth_msgs/$msg_hash/seen_by` does not include their
    address.
 2. It's correctly formatted.
 3. It's reached the required number of confirmations on the Ethereum chain
 
 Each event that a validator is voting to include must be individually signed by 
-them. The vote extension data field will be a Borsh-serialization of something 
-like the following.
+them. If the validator is not voting to include any events, they must still
+provide a signed voted extension indicating this.
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply documentation/spec/src/interoperability/ethereum-bridge.md:111]: Good idea.

**Review by batconjurer — APPROVED** [review]

File: documentation/spec/src/interoperability/ethereum-bridge.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (james-chf) [reply documentation/spec/src/interoperability/ethereum-bridge.md:114]: We will have to change the `VoteExtension` struct slightly (but won't update anything wrt. in this PR). Maybe something like

```rust
pub enum VoteExtension {
    Empty(BlockHeight),
    EthereumEvents(Vec<SignedEthEvent>)
}
```

[the remark this answers is not in the document]
Comment (james-chf) [reply documentation/spec/src/interoperability/ethereum-bridge.md]: I'm learning about vote extensions still and just realized they are signed by the validator's Tendermint node in any case, maybe this Tendermint-level signing is suitable for our purposes (although I don't see a way for us to get the Tendermint signature, if Tendermint hands us a vote extension from another validator then can we trust Tendermint has validated that it was in fact signed by that validator?). Will look more into this

**Review by batconjurer — APPROVED** [review]

**Pull request merged**